vmware-event-router/internal/processor/knative.go

```diff
@@ -0,0 +1,134 @@
+package processor
+
+import (
+	"context"
+	"crypto/tls"
+	"log"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/client"
+	"github.com/pkg/errors"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/color"
+	config "github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/config/v1alpha1"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/metrics"
+)
+
+const (
+	knativeDefaultRetryInterval = time.Second * 3
+	knativeDefaultMaxRetries    = 3
+)
+
+// knativeProcessor implements the Processor interface
+type knativeProcessor struct {
+	address       string
+	verbose       bool
+	retryInterval time.Duration
+	maxReTries    int
+	client        client.Client
+	*log.Logger
+	lock  sync.RWMutex
+	stats metrics.EventStats
+}
+
+func NewKnativeProcessor(ctx context.Context, cfg *config.ProcessorConfigKnative, ms metrics.Receiver, opts ...KnativeOption) (Processor, error) {
+
+	logger := log.New(os.Stdout, color.Purple("[Knative] "), log.LstdFlags)
+	kProcessor := knativeProcessor{
+		Logger:        logger,
+		retryInterval: knativeDefaultRetryInterval,
+		maxReTries:    knativeDefaultMaxRetries,
+	}
+
+	// apply options
+	for _, opt := range opts {
+		opt(&kProcessor)
+	}
+
+	if len(cfg.Address) > 0 {
+		kProcessor.address = cfg.Address
+	}
+
+	tlsConfig := &tls.Config{
+		//Certificates:       []tls.Certificate{cert},
+		//RootCAs:            clientCertPool,
+		InsecureSkipVerify: cfg.InsecureSSL,
+	}
+
+	httpTransport := &http.Transport{TLSClientConfig: tlsConfig}
+
+	// Create protocol and client
+	p, err := cloudevents.NewHTTP(cloudevents.WithRoundTripper(httpTransport))
+	if err != nil {
+		logger.Printf("failed to create protocol for knative client, %v\n", err)
```

Comment:
redundant log statements, just return the error

Reply:
Updated code as per the comments.